lib/Models/TerriaJsonCatalogFunction.js

```diff
@@ -443,4 +491,102 @@ TerriaJsonCatalogFunction.prototype.invoke = function() {
   return promise;
 };
 
+// private method split into function to make it recursive
+TerriaJsonCatalogFunction.prototype._handleHttp202 = function(all) {
+    //  A HTTP 202 response to try and get the data from polling acceptedUrl 
+    //    after a delay of acceptedDelay || 10 seconds
+    if (all.status !== undefined && all.status === 202) {
+    // else not a 202 response, all is text from Resource.fetch
+    // and it does not have a status property
+      if (this.acceptedUrl === undefined || 
+        typeof this.acceptedUrl !== "string") {
+          throw new TerriaError({
+            sender: this,
+            title: "Misconfigured init file",
+            message: "Service not configured for TerriaJsonCatalogFunction HTTP 202."
+          });
+      }
+     
+      // JSON-ify the body here, used Resorce.fetch, so body must be text
+      var jsonBody;
+      if (all.response !== undefined && typeof all.response === "string" &&
+          all.response !== "") {
+          jsonBody = JSON.parse(all.response);
+      }
+      if (typeof jsonBody === "object") {
+          all.response = jsonBody;
+      } else {
+        // add support for 202 bodies that are not json here if desired
+        // not returning here because acceptedUrl might not reference Body
+        // and this is not an error condition
+      }
+      var newResourceOpts = {
+        url: "",
+        responseType: "text",
+        headers: {
+          Accept: "application/json,*/*;q=0.01"
+        },
+        returnType: "XHRJSONHEADERS"
+      };
+      // ability to handle Accepted URL locations like first.second
+      // or Headers.location or Body.prop1.prop2.whatever.etc
+      // cant be a static input and 
+      // needs to be called to re-initialise the first next value,
+      var referenceIterator = this.acceptedUrl.split('.')[Symbol.iterator]();
```

Comment:
To better fit with the code style that the rest of Terria uses, would it be possible for you to use a for loop here instead of directly calling next on the iterators?

Reply:
Ok